src/test/pages/EventDetails.test.tsx

```diff
@@ -0,0 +1,230 @@
+import { TestUserRole, checkPageTitle, renderTestEnvironment } from "../utils";
+import { mockEvent } from "../../mocks/data";
+import { rest } from "msw";
+import { API_PATH } from "../../app/services";
+import { IsaacEventPageDTO } from "../../IsaacApiTypes";
+import { screen, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import * as eventServices from "../../app/services/events";
+import { formatAddress } from "../../app/components/pages/EventDetails";
+import { FRIENDLY_DATE_AND_TIME, TIME_ONLY } from "../../app/components/elements/DateString";
+
+const getButton = (buttonText: string) => {
+  const eventButton = screen.getByRole("button", { name: buttonText });
+  return eventButton;
+};
+
+const getFields = () => {
```

Comment:
This is not very efficient, and it's making the test unnecessarily slow, because you never care about more than two fields in one test, you normally only care about one anyway. I could be wrong, and it may be that the `queryBy` and `getBy` methods are not slow, but it's still strange to have a method that returns so many fields when you normally only care about one or two per test.

Reply:
Now that you've moved them to be functions, there is no need for `getFields`, you can have them as independent functions.